<a href="https://colab.research.google.com/github/annchirackal/Tensorflow-Developer-Certification/blob/TF_certification_practice_CNN/10_CNN_Computer_Vision_Multiclass_Classtification.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

The intension of this notebook is to do muticlass classification for images.
Our data consistes image of different food itens and we are going to bulid a model which can identify each of the food items

In [1]:
from tensorflow.keras import layers,optimizers
import os
from tensorflow.keras.utils import image_dataset_from_directory

import zipfile

In [2]:
# Download zip file of 10_food_classes images
# See how this data was created - https://github.com/mrdbourke/tensorflow-deep-learning/blob/main/extras/image_data_modification.ipynb
!wget https://storage.googleapis.com/ztm_tf_course/food_vision/10_food_classes_all_data.zip

--2024-01-09 14:09:20--  https://storage.googleapis.com/ztm_tf_course/food_vision/10_food_classes_all_data.zip
Resolving storage.googleapis.com (storage.googleapis.com)... 142.251.2.207, 2607:f8b0:4023:c0d::cf, 2607:f8b0:4023:c03::cf
Connecting to storage.googleapis.com (storage.googleapis.com)|142.251.2.207|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 519183241 (495M) [application/zip]
Saving to: ‘10_food_classes_all_data.zip’

10_food_classes_all 100%[===================>] 495.13M   209MB/s    in 2.4s    

2024-01-09 14:09:23 (209 MB/s) - ‘10_food_classes_all_data.zip’ saved [519183241/519183241]



In [3]:
zip_ref=zipfile.ZipFile("10_food_classes_all_data.zip")
zip_ref.extractall()
zip_ref.close()



In [4]:
for root, dirs, files in os.walk("10_food_classes_all_data"):
  print(f"There are {len(dirs)} directores and {len(files)} images in {root} ")

There are 2 directores and 0 images in 10_food_classes_all_data 
There are 10 directores and 0 images in 10_food_classes_all_data/test 
There are 0 directores and 250 images in 10_food_classes_all_data/test/chicken_curry 
There are 0 directores and 250 images in 10_food_classes_all_data/test/ice_cream 
There are 0 directores and 250 images in 10_food_classes_all_data/test/steak 
There are 0 directores and 250 images in 10_food_classes_all_data/test/grilled_salmon 
There are 0 directores and 250 images in 10_food_classes_all_data/test/pizza 
There are 0 directores and 250 images in 10_food_classes_all_data/test/ramen 
There are 0 directores and 250 images in 10_food_classes_all_data/test/fried_rice 
There are 0 directores and 250 images in 10_food_classes_all_data/test/sushi 
There are 0 directores and 250 images in 10_food_classes_all_data/test/chicken_wings 
There are 0 directores and 250 images in 10_food_classes_all_data/test/hamburger 
There are 10 directores and 0 images in 10_foo

So, we have 10 image classes and we have 750 training images and 250 test images in each  classes

In [5]:
test_path="10_food_classes_all_data/test"
train_path="10_food_classes_all_data/train"

In [6]:
train_data=image_dataset_from_directory(
    train_path,
  image_size=(224, 224),
  batch_size=32)



Found 7500 files belonging to 10 classes.


In [7]:
for images,labels in  train_data.take(1):
  print(images.shape,labels)

(32, 224, 224, 3) tf.Tensor([8 6 3 9 0 6 0 0 8 9 9 5 2 3 6 1 0 2 5 5 6 3 4 9 9 0 1 3 4 0 5 1], shape=(32,), dtype=int32)
